### ISSUE TITLE
V1 - cleanup from implementation requirements

### ISSUE THREAD

**Issue by elarlang** [OWASP/ASVS#2137]
TLDR from #1063:

  * We have changed the scope and removed requirements that are not clear about the application, such as software lifecycle, code beauty, etc.
  * We collect documentation requirements into V1 that are preconditions for implementing or testing the application security

Only requirements that need to be moved away from V1 are listed here.

---

**V1.1**

* [x] 1.1.4 - #1541
  * _Verify documentation and justification of all the application's trust boundaries, components, and significant data flows._
* [x] 1.1.5
  * #1541 
  * #993
  * _Verify definition and security analysis of the application's high-level architecture and all connected remote services._
* [x] 1.1.6 - #1434
  * _Verify implementation of centralized, simple (economy of design), vetted, secure, and reusable security controls to avoid duplicate, missing, ineffective, or insecure controls._

---

**V1.2** - only documentation requirements

---

**V1.3** - nothing yet

---

**V1.4**

* [x] 1.4.4 - implementation requirement, or code beauty? ~Feels a bit duplicated discussion to V1.1.6 (#1434)~ Opened #2147 to discuss as doesn't feel like a requirement.
  * _Verify the application uses a single and well-vetted access control mechanism for accessing protected data and resources. All requests must pass through this single mechanism to avoid copy and paste or insecure alternative paths._
* [x] 1.4.5 - #1183
  * _Verify that attribute or feature-based access control is used whereby the code checks the user's authorization for a feature or data item rather than just their role. Permissions should still be allocated using roles._

---

**V1.5**

V1.5.1 can be more clear towards documentation

---

**V1.6**

None of those is a documentation requirement, all are implementation requirements and must belong to V6.

* [ ] 1.6.1
  * _Verify that there is an explicit policy for management of cryptographic keys and that a cryptographic key lifecycle follows a key management standard such as NIST SP 800-57._
* [ ] 1.6.2
  * _Verify that consumers of cryptographic services protect key material and other secrets by using key vaults or API based alternatives._
* [ ] 1.6.3 - #1544
  * _Verify that all keys and passwords are replaceable and are part of a well-defined process to re-encrypt sensitive data._
* [ ] 1.6.4
  * _Verify that the architecture treats client-side secrets (such as symmetric keys, passwords, or API tokens) as insecure and never uses them to protect or access sensitive data._

---

**V1.7** - ok, only documentation requirements

---

**V1.8** - ok, only documentation requirements

---

**V1.9** - ok, nothing there

---

**V1.10** - ok, nothing there

---

**V1.11**

* [x] 1.11.2 - #1303
  * _Verify that all application flows including authentication, session management and access control, maintain a consistent application and user state to prevent race conditions and business logic flaws._
* [x] 1.11.3 - #2138
  * _Verify that all high-value business logic flows, including authentication, session management and access control are thread safe and resistant to time-of-check and time-of-use race conditions._

---

**V1.12** - ok, nothing there

---

**V1.13** - ok, nothing there

---

**V1.14**

* [x] 1.14.5 - #1531
  * _Verify that application deployments adequately sandbox or isolate at the network level to delay and deter attackers from attacking other applications, especially when they are performing sensitive or dangerous actions such as deserialization._
* [x] 1.14.7 - #1389
  * _Verify that the application is able to discern and utilizes the user's true IP address to provide for sensitive functions, including rate limiting and logging._

**Comment by tghosth** [comment]
Comments:

* I think that 1.6.1 is a classic documentation requirement but might need rewording.
* The rest of 1.6 will need to be merged into V6 when we have the draft from Daniel.


All the rest I think I have commented on the actual issues

**Comment by elarlang** [comment]
This issue can be closed. The cleanup is done and requirements in V1.6 / V6 are reviewed separately.
